src/main/java/com/exasol/dbbuilder/dialects/exasol/AdapterScript.java

```diff
@@ -53,7 +49,129 @@ public String getContent() {
         return this.content;
     }
 
+    /**
+     * Get the debugger connection for this adapter script.
+     * 
+     * @return debugger connection
+     */
+    public Optional<String> getDebuggerConnection() {
+        return this.debuggerConnection;
+    }
+
     public enum Language {
         JAVA, PYTHON, LUA, R
     }
-}
\ No newline at end of file
+
+    /**
+     * Get a builder for {@link AdapterScript}.
+     * 
+     * @return builder for {@link AdapterScript}
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Builder for {@link AdapterScript}.
+     */
+    public static final class Builder {
+        protected Schema parentSchema;
+        protected String name;
+        protected boolean owned;
+        private ExasolImmediateDatabaseObjectWriter writer;
+        private Language language;
+        private String content;
+        private Optional<String> debuggerConnection = Optional.empty();
```

Comment:
Don't use `Optoinal` as field. It does not prevent you from needing to check for `null`.  Null checking just get's more complicated. There are lots of good articles on the Internet, that explain the pitfalls. It's really designed as return value wrapper.